c10/core/DispatchKeySet.h

```diff
@@ -123,7 +123,7 @@ static inline DispatchKey legacyExtractDispatchKey(DispatchKeySet s) {
   // top of existing "normal" keys like CPU/CUDA, you need to add it
   // here.  At the moment, RequiresGrad (replacement for Variable)
   // is the most likely key that will need this treatment.
-  return s.highestPriorityTypeId();
+  return s.remove(DispatchKey::BackendSelect).highestPriorityTypeId();
```

Comment:
perhaps you should update the comment to explain why BackendSelect is being removed.

Reply:
Added a comment.

Reply:
I know it's special -- but the question was why it's special in a different way than Variable (i.e. why BackendSelect and Variable aren't always inserted/removed in the same places).  Can you add a comment to that effect?

Reply:
@ezyang, @gchanan does it make sense? Im having a hard time phrasing this.